File: app/models/article.rb
# == Schema Information
#
# Table name: articles
#
#  id             :integer          not null, primary key
#  headline       :string(255)
#  lead_paragraph :string(255)
#  pubdate        :datetime
#  word_count     :integer
#  created_at     :datetime
#  updated_at     :datetime
#

class Article < ActiveRecord::Base
  has_many :taggings
  has_many :keywords, through: :taggings, source: :keyword
end

File: app/assets/javascripts/models/article.js
// for more details see: http://emberjs.com/guides/models/defining-models/

AutocompleteRedis.Article = DS.Model.extend({
  headline:       DS.attr("string"),
  leadParagraph:  DS.attr("string"),
  pubdate:        DS.attr("string"),
  wordCount:      DS.attr("date"),

  keywords: DS.hasMany("keyword", {
    async: true
  })
});
